src/main/java/org/opensearch/knn/index/remote/RemoteIndexClient.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.remote;
+
+import org.opensearch.cluster.metadata.RepositoryMetadata;
+import org.opensearch.index.IndexSettings;
+import org.opensearch.knn.index.codec.nativeindex.model.BuildIndexParams;
+
+import java.io.IOException;
+
+/**
+ * Interface for a Remote Index Client. This will support future implementations for protocols such as gRPC.
+ */
```

Comment:
```suggestion
/**
 * Interface which dictates how we interact with a remote index build service
 */
```

Or something like that. Don't mention future implementations especially if they're not immediately planned.